Add new admin UI as beta

Arnei opened the pull request:
This PR adds the new Admin UI as optional feature alongside the old Admin UI. This allows users to the test out the new Admin UI and easily switch back the old Admin UI if they find the new one to be lacking in some regard.

Much like Studio and the Editor, the code for the new Admin UI lives in its own repository https://github.com/opencast/opencast-admin-interface (well not quite yet, https://github.com/opencast/opencast-admin-interface/pull/134 still needs to be merged). The new Admin UI will be built and released in the new repository as well, Opencast will just need to download and unpack the build. This should make development of the new Admin UI easier, and the build process for Opencast shorter.

This PR also which modules are responsible for login. `runtime-info-ui` is now the single module responsible for deploying the login page, the page in `admin-ui-frontend` goes unused. This allows us to get rid of the `runtime-info-ui-ng` module and hopefully lessen confusion regarding the login page. After logging in, a user will be redirected to the Admin UI if on the admin node and to the Rest Docs if on presentation or worker node.

**UPDATE: https://github.com/opencast/opencast-admin-interface/pull/134 has been merged, this PR is ready now!**

Commit:
Merge pull request #174 from TIK-NFL/picard-user-details

Add user, group and ACL details modals

Commit:
resolved merge conflicts

Commit:
replaced console.log and modal  navigations in files added after merge

Commit:
changed form to formik form in login page

Commit:
adjusted imports after merge with refactoring branch

Commit:
Merge branch 'admin-ui-picard' into picard-login

Commit:
fixed login layout and added pom for future building via maven

Commit:
updated package-lock.json

SimoneSchmidt committed:
Event Details Workflow Tab: logic for actions added

Commit:
added publications tab in event details

SimoneSchmidt committed:
Workflow Tab of Event Details: Actions finished and Link-logic added

Finished adding the logic of normal actions and delete action for workflows.
Added the logic for the link to the workflow details sub-tab.

SimoneSchmidt committed:
Merge pull request #194 from TIK-NFL/picard-event-details-publication

Publications tab in event details modal

SimoneSchmidt committed:
Unnecessary Log removed

Commit:
changes requested in PR #195

SimoneSchmidt committed:
Merge pull request #195 from TIK-NFL/picard-login

Login Page

SimoneSchmidt committed:
Resolved merge conflicts

SimoneSchmidt committed:
Resolved merge conflicts

SimoneSchmidt committed:
Merge branch 'picard-event-details' into admin-ui-picard

Merge branch 'picard-event-details', containing the implementation for the event details workflow tab, after resolving merge conflicts

pascalseeland committed:
Merge remote-tracking branch 'upstream/develop' into develop

pascalseeland committed:
Merge branch 'develop' of github.com:TIK-NFL/opencast into develop

pascalseeland committed:
Merge remote-tracking branch 'upstream/develop' into develop

Commit:
configured pom of admin-ui-picard to be built with rest of opencast

Commit:
Compile error caused by typo, summary page of themes wizard and row counter in group table fixed

Commit:
resolved merge conflicts

Commit:
put login in own project cause of redirecting of backend

Commit:
login app starts and renders login page

Commit:
configured login app to use mockbackend and backend

Commit:
ejected login app; added pom.xml to bundle in build to login app; fixed pom.xml of main app and login app to be really build during maven build

SimoneSchmidt committed:
Workflow Tab of Event Details finished

Finished logic for Sub-Tab "Workflow Details".
Added views and logic for Sub-Tabs "Operations", "Operation Details", "Errors" and "Error Details".

SimoneSchmidt committed:
comments added

code comments added to html sections for Event Details Comments Tab

Commit:
deleted extra login app and configured admin-ui to use runtime-info-ui for login; fixed request paths in axios

Commit:
added logout functionality

SimoneSchmidt committed:
Metadata Tab of Event Details finished

Added views and logic for Metadata Tab of Event Details Modal.
Fixed wrong button labels for Access Policy Tab and Workflows Tab of Event Details Modal.

SimoneSchmidt committed:
Merge pull request #344 from TIK-NFL/picard-debugging

Compile error caused by typo, summary page of themes wizard and row c…

SimoneSchmidt committed:
Merge branch 'admin-ui-picard' into picard-workflow-details

Merge to include bug fixes

SimoneSchmidt committed:
Merge branch 'admin-ui-picard' into picard-event-details-metadata

Merge to include bug fixes.

SimoneSchmidt committed:
Changes requested in PR #341

Performed changes requested for PR #341.

Commit:
Merge pull request #341 from TIK-NFL/picard-event-details-metadata

Metadata Tab of Event Details finished

SimoneSchmidt committed:
Changes requested in PR #332

Added the changes requested for PR #332.

SimoneSchmidt committed:
Fixed style warnings

Fixed html nesting in hierarchy navigation for event details workflow details which caused warnings.

Commit:
Merge branch 'admin-ui-picard' into picard-workflow-details

Commit:
Merge pull request #332 from TIK-NFL/picard-workflow-details

Picard event details: workflow sub-tabs

SimoneSchmidt committed:
Merge pull request #340 from TIK-NFL/picard-login

login functionality and built with new UI

Commit:
ui elements are only rendered if user has the appropriate role

Commit:
role check in main navigation

Commit:
fixed bugs in appearence caused by introduction of roles

Commit:
fixed role cell in user page; now roles are displayed in mulitple lines instead of one

Commit:
added embedding code modal

Commit:
removed logs in embedding code modal and started implementation of adopter registration modal

Commit:
Changes according review of PR#353

Commit:
Merge branch 'develop' of https://github.com/TIK-NFL/opencast into develop

Commit:
resolved merging confilicts

SimoneSchmidt committed:
Added Asset Tab for Event Details Modal

Added Asset Tab for Event Details Modal with all sub-Tabs.
Only logic for file upload and video preview player still missing.

SimoneSchmidt committed:
changes according to review of PR#387

SimoneSchmidt committed:
Merge pull request #387 from TIK-NFL/picard-event-details-assets

Added Asset Tab for Event Details Modal

SimoneSchmidt committed:
Merge branch 'admin-ui-picard' into picard-role-management

Commit:
Merge pull request #353 from TIK-NFL/picard-role-management

Role Management

Commit:
resolved merge conflicts

SimoneSchmidt committed:
Added Extended Metadata Tabs for Details Modals

Added the Extended Metadata Tabs for Event Details Modal and Series Details Modal.
Minor refactoring.
Added role management to Event Details Modal Assets Tab.

Commit:
implemented terms of use page and added test data for registration modal

SimoneSchmidt committed:
Implemented requested changes for PR#393

Commit:
Merge pull request #393 from TIK-NFL/picard-event-details-extended-metadata

Added Extended Metadata Tabs for Details Modals

Commit:
finished implementation of adpoter registration modal and added comments

Commit:
changes according PR#370

Commit:
removed console log

Commit:
Merge pull request #370 from TIK-NFL/picard-embedding-code

Embedding Code Modal and Adopter Registration Modal

Commit:
implemented Hotkey Cheat Sheet and began to add hotkeys

Commit:
fixed build error

Commit:
resolved merge conflicts

SimoneSchmidt committed:
Added Scheduling Tab of Event Details Modal

Added views and logic for Scheduling Tab of Event Details Modal.

Commit:
added remaining hotkeys

Commit:
Merge branch 'admin-ui-picard' into picard-debugging

Commit:
constants containing path to media module, rest documentation and general documentation removed. Now they are queried from redux state, since these paths are part of infomation in me.json

Commit:
Footer now queuries backend for version on start up

Commit:
redirect to services table page when clicking on services in health status

Commit:
disable active bulk actions button when switching from event table page to series table page and back

Commit:
moved hardcoded constants to config files

SimoneSchmidt committed:
Implemented changes requested on PR#422

SimoneSchmidt committed:
Merge branch 'admin-ui-picard' into picard-event-details-scheduling

SimoneSchmidt committed:
Merge pull request #422 from TIK-NFL/picard-event-details-scheduling

Event details scheduling tab added

Commit:
changes according review in PR#423

Commit:
Merge pull request #423 from TIK-NFL/picard-hotkeys

Hotkeys

Commit:
changes according to review of PR#437

SimoneSchmidt committed:
Import Bugfix

Fixed bug caused by one import in EventDetails.js vanishing during one of the last merges.

SimoneSchmidt committed:
Rearanged Imports

Rearanged order of Imports for better readability.

Commit:
Merge branch 'admin-ui-picard' into picard-debugging

Commit:
Merge pull request #437 from TIK-NFL/picard-debugging

Some minor things and bug fixes

SimoneSchmidt committed:
Merge pull request #446 from TIK-NFL/picard-import-bugfix

Import Bugfix in EventDetails.js

SimoneSchmidt committed:
Started Statistics Tab of Event Details Modal

Created view (html) for Statistics Tab of Event Details Modal.

Commit:
basic functionality of workflow config panel in add event wizard

Commit:
default workflow configuration gets loaded in dorpdown of new event wizard

Commit:
finished basic functionality of workflow configuration panel in new event wizard

Commit:
fixed bugs occured when testing with proxy to oc instance; updated post event method

Commit:
updated workflow config panel in event details

Commit:
updated workflow configuration panel in start-task bulk action

Commit:
added possiblity for user to disable config options in workflow config panels

Commit:
added new xml tag in workflows for workflow panel in json and adjusted backend accordingly

Commit:
implemented add assets button in event details

Commit:
added preview player for media in event details

Commit:
fixed timezone issue and dropdown menus in scheduling page in new event wizard

Commit:
adjusted save buttons in series details

SimoneSchmidt committed:
Finished views for Statistics Tab of Event Details Modal

Added BarChart Conponent for displaying time series statistics data as bar chart.
Added logic for interaction with formik inputs for time series statistics (API requests for changing displayed data still missing).

Commit:
selection of columns in main table pages do not change anymore when leaving page or fetching resources

Commit:
selection of rows in events and series table page are not reset when fetching resources anymore

SimoneSchmidt committed:
Finished Statistics Tab of Event Details Modal

Added logic for value update API calls.
Added comments.

Commit:
removed unused imports

Commit:
began to implement extended metadata page in new event/series wizard; not usable yet because of need of further information

Commit:
fixed override of filters and added link to filtered events when click on series or location name in corresponding table view

SimoneSchmidt committed:
Implemented Changes requested on PR#496

Commit:
changes according review PR#484

SimoneSchmidt committed:
Revert accidentally changed package-lock.json files of unchanged modules to earlier version.

Commit:
changes according to review #481

Commit:
Merge pull request #496 from TIK-NFL/picard-event-details-statistics-tab

Picard event details statistics tab

Commit:
resolved merging conflicts

Commit:
Merge branch 'picard-event-details' into admin-ui-picard

Commit:
resolved merge conflict

Commit:
Merge branch 'picard-workflowpanel' into admin-ui-picard

SimoneSchmidt committed:
Added Statistics Tab of Series Details Modal

Performed refactoring of existing statistics code and imports.
Added Statistics Tab to Series Details Modal

Commit:
Merge branch 'admin-ui-picard' into picard-debugging

Commit:
fixed check of acls in acl details and removed some unused imports

Commit:
fixed creation of series with theme

SimoneSchmidt committed:
Added Statistics Page

Added Statistics Page, which displays statistics for the organization.

SimoneSchmidt committed:
Implemented Changes requested on PR#508

SimoneSchmidt committed:
Fixed style bug.

SimoneSchmidt committed:
Merge pull request #508 from TIK-NFL/picard-statistics-tabs-and-pages

Remaining statistics tabs and pages added.

SimoneSchmidt committed:
Linked Series Details in Event Action Cell

Added the functionality for opening Series Details, when clicking the "open series details" button in Event Action Cell.

Commit:
focus on first field/input when opening wizards

SimoneSchmidt committed:
Merge pull request #509 from TIK-NFL/picard-event-action-cell-series

Linked opening of Series Details Modal in "open series details" button in Event Action Cell

Commit:
updated assets page in new event wizard

Commit:
changes according review of PR#497

SimoneSchmidt committed:
Merge branch 'admin-ui-picard' into picard-debugging

SimoneSchmidt committed:
Merge pull request #497 from TIK-NFL/picard-debugging

Fixed Some Bugs and Added Some Minor Features

Commit:
reset filter when click on a stats filter

Commit:
Merge branch 'picard-debugging' of https://github.com/TIK-NFL/opencast into picard-debugging

Commit:
Merge branch 'admin-ui-picard' into picard-debugging

Commit:
fixed system notification banner

Commit:
set filter for failed backend services when clicking on backend services in system notification dropdown

SimoneSchmidt committed:
Fixed Bug in Metadata Tabs

Previously in Metadata Tabs for fields which have drop-downs for value selection, the key of the value was displayed for the selected value instead of the name of the value. This has been fixed, now the translated name of the selected value is displayed.

Commit:
formatted period of start date filter

SimoneSchmidt committed:
Fixed Bug no Metadata for Events with Status "Running"

Previously for events with status "Running" no metadata was displayed in event details. This has been fixed.

Commit:
fixed bug that end date in table filter needed to pick twice

SimoneSchmidt committed:
Fixed Bug in Metadata Tabs - additional fix

Previously in Metadata Tabs for fields which have drop-downs for value selection, while they are editable, the key of the value was displayed for the selected value instead of the name of the value, when in non-editable mode. This has been fixed, now the translated name of the selected value is displayed, even, when the field is non-editable.

Commit:
fixed redirect to /events/events; removed some unused imports

Commit:
fixed stats period filters of event table view

Commit:
updated react-router from v5 to v6

SimoneSchmidt committed:
Adapted Asset Upload

The options for the asset upload in event details and new event wizard, as well as source upload in new event wizard, are now fetched via API from the opencast configuration. The new event summary page has also been adapted to properly display the names of the asset and source types.

SimoneSchmidt committed:
Adapted Asset Upload again

The Asset Upload step/tab in New Event Wizard and the Add Assets Button in the Assets Tab of Event Details Modal are now only shown, when there are asset upload options, which were returned from the API. EVENTS.EVENTS.NEW.UPLOAD_ASSET.WORKFLOWDEFID from API is now used in additional asset upload. Minor fixes.

Commit:
fixed crash caused by use of browser back button or hotkey

SimoneSchmidt committed:
Implemented requested changes on PR#512

Commit:
transformed the stepper navigation in wizards to a stepper button navigation; now the user can not only navigate trough the buttons at the buttom but also via click on the desired page on the stepper; the stepper button navigation also validates the input and only works if the current page contains valid inputs

Commit:
Merge pull request #512 from TIK-NFL/picard-bugfixes

Bugfixes

Commit:
set pull interval from every minute to every 5 sec; fixed update after applying filters trough link in tables

SimoneSchmidt committed:
Added Extended Metadata Page to Resource Wizards

Added Extended Metadata Page to New Event and New Series Wizards.
Added Extended Metadata section to Wizard Summary pages.
Adapted eventThunks and seriesThunks, to get extended metadata info out of API request and to add extended metadata values to API requests for creating new resources.
Removed some unnecesarry imports.

SimoneSchmidt committed:
Merge branch 'admin-ui-picard' into picard-bugfixes

Commit:
fixed fetching of filters when browser back button is clicked

Commit:
fixed that stepper icon fills before page is checked for validity

Commit:
Changes accorging review of PR#513

Commit:
changed rendering of blue box in table filters to show the actual label of the filter value

SimoneSchmidt committed:
Fixed feedback-link in footer

Url value for the feedback-link in the site-footer is now the value from the org.properties from the user info, fetched via API.

Commit:
changes and fixes according review of PR#513

Commit:
Changes according review of PR#513

Commit:
resolved merge conflicts

Commit:
Merge pull request #513 from TIK-NFL/picard-debugging

Some Bug Fixes

SimoneSchmidt committed:
Added check in Delete Series Dialog

Added checks for:
    - does the series contain events
    - is deletion of series containing events allowed
to opening of Delete Series Dialog.
If deletion is allowed and the series contains events, a warning is now displayed.
If deletion is not allowed an error message is now displayed and deletion is not possible (no delete button).

SimoneSchmidt committed:
Adapted source option visibility

In source page of New Event Wizard,  source option visibility was adapted. Selecting the source modes "SCHEDULE_SINGLE" and "SCHEDULE_MULTIPLE" is now only possible and they are only shown, if capture agents exist and the user has the corresponding access role(s) to use (at least one of) them.

Commit:
fixed disabled next button after click on previous button

Commit:
fixed scheduling of mulitple repeating events and try to improve navigation in wizard via stepper buttons

Commit:
Summary page in wizards now only directly reachable via stepper if other pages are valid

Commit:
fixed validation of wizard pages if previous buttons was clicked

Commit:
fixed statement of check for reachablility of summary page in wizards

Commit:
adjusted uplaod of files in source page of new event wizard to support upload of multiple files

SimoneSchmidt committed:
Adapted capture agent drop-downs

In new event wizard, edit scheduled events modal and event details modal, the capture agent drop-downs have been adapted. Now only capture agents for which the user has access rights are available for selection in the drop-downs.
Fixed event selection in edit scheduled events modal, so that the user has to have access rights for the used capture agent in order to be able to select it.
Some slight adjustments of source option visibility in new event wizard.
Fixed empty available options in some drop-downs, they are now gone.
Slight refactoring.

Commit:
added a reload of page after successful logout to fix redirect

Commit:
added link to standalone video editor in action cell

Commit:
found workaround for redirect after login for now; the current fix is not optimal however now the login can be used without changing the url after a successful login by hand

Commit:
commiting change that was not added in commit before; see last commit for further information

Commit:
merge changes accidently made on main branch

Commit:
removed unused player hotkeys

Commit:
implemented filter in technical date cell of events

Commit:
added notification for error message if user information can not be loaded

Commit:
added notification to main table view; now error notifications are also shown above the table

SimoneSchmidt committed:
Fixed Bulk Schedule Update

On the edit page of the udate scheduled event modal, the series dropdown and the weekday selection have been fixed.
On the summary page of the modal, the way series name and weekday are displayed have been fixed.
The request data for the API request for updating the event schedules has been adapted, so that the metadata and schedule are properly updated.

SimoneSchmidt committed:
Changes according to first review of PR #537

SimoneSchmidt committed:
Changes according to second review of PR #537

SimoneSchmidt committed:
Merge branch 'admin-ui-picard' into picard-bugfixes

SimoneSchmidt committed:
Merge pull request #537 from TIK-NFL/picard-bugfixes

Some bugfixes:
  - Added Extended Metadata Page to Resource Wizards
  - Fixed feedback-link in footer
  - Added check in Delete Series Dialog
  - Adapted source option visibility
  - Adapted capture agent drop-downs
  - Fixed Bulk Schedule Update

Commit:
added and adjusted some notifications

Commit:
resolved merge conflicts

Commit:
changes according review of PR#549 and fixed dropdown menu of locations in new events wizard

Commit:
added remaining notifications and removed unused notification translation keys

Commit:
removed some unused notification translation keys

SimoneSchmidt committed:
Implemented searchable drop-down

Implemented searchable drop-down for the drop-down selection of RenderField Component.
Some minor bug-fixes:
 - in New Resource Summary pages: wrong display of collection values as keys instead of label fixed
 - in bulk schedule update and start task: bugs caused by change in WizardStepper Component fixed

Commit:
changes according review of PR#549

SimoneSchmidt committed:
Merge pull request #549 from TIK-NFL/picard-debugging

Multiple Bug Fixes

SimoneSchmidt committed:
Changes according to review of PR #563

SimoneSchmidt committed:
Resolved merge conflict in package-lock.json file

SimoneSchmidt committed:
Merge pull request #563 from TIK-NFL/picard-bugfixes

Implemented searchable drop-down

SimoneSchmidt committed:
Fixed top Navigation Buttons in Bulk Action Modals

Fixed the way the WizardStepper is integrated in StartTaskModal and EditScheduledEventsModal, so that the navigation buttons on top work as intended.

SimoneSchmidt committed:
Fixed small bug in event selection of EditScheduledEventsModal, that occurred, when changing the selection again, after having visited another tab.

SimoneSchmidt committed:
Changed capture agent drop-downs to searchable drop-downs.

SimoneSchmidt committed:
New drop-down in EventDetailsSchedulingTab + Requested Changes

Changed time selection in Scheduling Tab of Event Details to new searchable drop-downs.
Small bug-fixes.
Changes according to review of PR #566.

SimoneSchmidt committed:
Removed some unused imports.

SimoneSchmidt committed:
Merge pull request #566 from TIK-NFL/picard-bugfixes

Bug Fixes:
 -   Fixed the way the WizardStepper is integrated in StartTaskModal and EditScheduledEventsModal, so that the navigation buttons on top work as intended.
-    Changed capture agent drop-downs to searchable drop-downs.
-    Changed time drop-downs in EventDetailsSchedulingTab to searchable drop-downs.

SimoneSchmidt committed:
Added remaining searchable drop-downs

Changed remaining drop-down menus, which are supposed to be searchable to the new searchable DropDown component (in create event, series and access policy wizards and event, series and access policy details modals).
Did slight refactoring and fixed code style at some places.

SimoneSchmidt committed:
Several bug-fixes

Changed drop-down for workflow in bulk start task modal to searchable.
Fixed pagination problem in tables.
Fixed faulty scheduling conflict check in select source page of create event wizard and scheduling tab of event details modal.

SimoneSchmidt committed:
Bug-fixes Studio link and Help menu

Fixed the Opencast Studio Link in the Header.
Fixed the text overflow for long entry names in the Help menu in the Header.

SimoneSchmidt committed:
Bug-fix Presenter filter

The presenter table filter, which is activated by clicking on a presenter name in the event table, has been fixed and now works properly.

SimoneSchmidt committed:
Merge branch 'picard-bugfixes' into admin-ui-picard

 - Added remianing searchable drop-downs
 - Fixed pagination problem in tables
 - Fixed faulty scheduling conflict check in select source page of create event wizard and scheduling tab of event details modal
 - Fixed the Opencast Studio Link in the Header
 - Fixed the text overflow for long entry names in the Help menu in the Header
 - Fixed the presenter filter for the events table
 - Slight refactoring

SimoneSchmidt committed:
Removed now unused file modules/admin-ui-picard/app/src/configs/utilsConfig.js

SimoneSchmidt committed:
Added and executed mvn spotless plugin for automatic style checks.

Arnei committed:
Fix empty statistics page

Without a statistics provider, the "Statistics" page only
shows a blank page. This is because one of the related
redux states does not get properly set in case of an
empty array. This fixes that.

Arnei committed:
Merge remote-tracking branch 'opencast-community/develop' into admin-ui-picard-develop-23-01-2023

Arnei committed:
Fix typo

Arnei committed:
Fix Picard package version

Arnei committed:
Remove everything ActiveMQ related from Picard

Arnei committed:
Fix table column names

Fixes various table column names for events and series that have changed over the years. This should make sorting for those columns work again.

Arnei committed:
Fix DateTimePicker crashing the page

Clicking on the start date field in the events details
metadata view would crash the page. This fixes that.

It also removes inadequate formatting. While this
still isn't optimal, I couldn't figure out how to get
dateTime formats from `language.json` to the
MUI components.

Arnei committed:
Make runtime-info-ui the universal login page

Turns the login page served by the runtime-info-ui module into the universal login page for all Opencast nodes.
This means the same login page is shown across nodes, instead of the admin node having a special login page.

Gets rid of the runtime-info-ui-ng module, which is no longer needed.

Gets rid of the overview page shown after logging in on non-admin nodes (on the root path). Instead, immediately redirects to the rest docs.

ToDos:
- Bring the login page up to the style, functionality and code of the login page in admin-ui-frontend
- Remove the login page from admin-ui-frontend

Arnei committed:
Fixes ESLint Warnings

Mostly fixes "react-hooks/exhaustive-deps" and "no-unused-vars".
Should be straightfoward.

Arnei committed:
Fixes ESLint Warnings

Mostly fixes "react-hooks/exhaustive-deps" and "no-unused-vars".
Should be straightfoward.

Arnei committed:
Reenable redux persist

This re-enable redux-persist, which was disabled for
debugging purposes. I'm leaving the comments about
debugging in, in case they come up useful later.

Arnei committed:
Fix table sorting mismatch between local and redux

Fixes an issue where the table sorting state would get set
wrongly in redux, due the a useState setter hook
triggering too late.

Arnei committed:
Merge pull request #38 from opencast/picard-fix-empty-statistics-page

Fix empty statistics page

Arnei committed:
Merge pull request #39 from opencast/admin-ui-picard-develop-23-01-2023

Admin UI picard develop 23 01 2023

Arnei committed:
Merge branch 'admin-ui-picard' into fix-datetimepicker-crashing

Arnei committed:
Merge pull request #41 from opencast/fix-datetimepicker-crashing

Fix DateTimePicker crashing the page

Arnei committed:
Merge branch 'admin-ui-picard' into fix-table-sorting-names

Arnei committed:
Merge branch 'admin-ui-picard' into fix-big-amount-of-eslint-warnings

Arnei committed:
Merge branch 'admin-ui-picard' into enable-redux-persist

Arnei committed:
Merge branch 'admin-ui-picard' into make-runtime-info-ui-the-universal-login-page

Arnei committed:
Merge pull request #40 from opencast/fix-table-sorting-names

Fix table column names

Arnei committed:
Merge pull request #102 from opencast/enable-redux-persist

Reenable redux persist

Arnei committed:
Merge pull request #99 from opencast/fix-big-amount-of-eslint-warnings

Fix big amount of eslint warnings

Arnei committed:
Merge pull request #96 from opencast/make-runtime-info-ui-the-universal-login-page

Make runtime-info-ui the universal login page

Arnei committed:
Turn admin-ui-picard into admin-ui-interface

Much like studio and editor, the new admin ui will not be built during the general Opencast build process, but builds will be pulled from a secondary repository.
This commit achieves that by removing the module containing the new
admin ui and replacing it with a module that only contains a pom.xml
that pulls the builds from opencast/admin-ui-interface

Arnei committed:
Allow runtime-info-ui to redirect to either admin UI

Introduces a new config file for the runtime-info-ui module, so users
can configure which admin ui they want to redirect to (or alternatively
a different subpath entirely)

Arnei committed:
Add back rest_docs

Apparently the rest_docs were deleted alongside the runtime-info-ui-ng
module at some point. This commit adds the files back to Opencast,
albeit to the runtime-info-ui module instead.

Arnei committed:
Add documentation on the new Admin UI

Added a file to the adopter docs that informs about the new Admin UI, how to accces it and how it is still in beta.

Arnei committed:
Merge remote-tracking branch 'opencast/develop' into add-new-admin-ui-as-beta

Arnei committed:
Fix new admin ui documentation page not showing

Was not linked in the relevant index.md files

Arnei committed:
Fix tests in admin-ui for configuration_panel_json

Test file jsons did not include data for configuration_panel_json
objects

Arnei committed:
Fix test in ffmpeg, missing bundle context

Mocks a bundle context for the EncodingProfileScanner in composer-ffmpeg

mtneug commented on etc/security/mh_default_org.xml:
```diff
@@ -29,12 +29,13 @@
     <sec:intercept-url pattern="/img/**" access="ROLE_ANONYMOUS" />
-    <sec:intercept-url pattern="/js/**" access="ROLE_ANONYMOUS" />
+    <sec:intercept-url pattern="/scripts/**" access="ROLE_ANONYMOUS" />
+    <sec:intercept-url pattern="/styles/**" access="ROLE_ANONYMOUS" />
```
Why the new paths? Can everything beloging to the admin UI be prefixed with `/admin-ui/...`?

Reason: simple loadbalancer rules when routing HTTP requests. I plan to run the admin UI as separate containers.

Arnei replied:
Those are for the `runtime-info-ui` module, which contains the login page and the rest docs, and is also responsible for redirecting to the `/admin-ui/index.html` after login.

I tried to explain why the login page is in a different module from the admin ui in the initial post. Also, I got very confused by Spring Security authentication and redirect methods and therefore opted what seemed to me like the simplest solution, resulting in these paths.

Arnei committed:
Merge remote-tracking branch 'upstream/develop' into add-new-admin-ui-as-beta

Arnei committed:
Re-Add package.json

Accidentally removed necessary package.json in runtime-info-ui. Now
readded.

Arnei committed:
Merge remote-tracking branch 'upstream/develop' into add-new-admin-ui-as-beta

# Conflicts:
#	.github/dependabot.yml
#	modules/runtime-info-ui-ng/package-lock.json
#	modules/runtime-info-ui-ng/package.json

Arnei committed:
Merge remote-tracking branch 'upstream/develop' into add-new-admin-ui-as-beta

# Conflicts:
#	modules/runtime-info-ui-ng/package-lock.json
#	modules/runtime-info-ui-ng/package.json

Arnei committed:
Update release

gregorydlogan commented on modules/admin-ui-interface/pom.xml:
```diff
@@ -0,0 +1,61 @@
+    <properties>
+        <opencast.basedir>${project.basedir}/../..</opencast.basedir>
+        <checkstyle.skip>false</checkstyle.skip>
+        <interface.url>https://github.com/opencast/opencast-admin-interface/releases/download/2023-04-12-2/oc-admin-ui-2023-04-12.tar.gz</interface.url>
```
How is this kept up to date as the other repo changes?  Same way as Studio - manually?

Arnei replied:
Pretty much, yeah.

Arguably it would be better to have a branch and release structure more similar to Opencast, so you could more easily backport changes, but for now I went with a way I know how to set up ^^'

gregorydlogan replied:
That's fair.  There's a conversation to be had in the community in terms of how this *should* work for all three modules.

gregorydlogan commented on etc/workflows/duplicate-event.xml:
```diff
@@ -85,6 +85,21 @@
       </script>
     ]]>
   </configuration_panel>
+  <configuration_panel_json>
```
There are a bunch of fields which show up in the old admin ui, but not the new one for this workflow (which only shows the number of dups, as defined here).  This causes things to be unhappy, but work:
```
2023-04-20 15:23:01,481 | INFO  | (DuplicateEventWorkflowOperationHandler:281) - Creating 4 new media packages from media package with id 0e583a0f-d1bc-450d-8f85-23aa5d95d176.
2023-04-20 15:23:01,487 | WARN  | (DuplicateEventWorkflowOperationHandler:371) - Could not parse: ${startDateTime} as date time
2023-04-20 15:23:01,488 | WARN  | (DuplicateEventWorkflowOperationHandler:373) - Using original event date Thu Apr 20 14:58:00 CST 2023 as default
```

Arnei replied:
It seems the new Admin UI currently only supports workflow config fields of types `radio`, `checkbox`, `number` and `text`. So I can't just add field of type `datetime-local` to the `configuration_panel_json`, as it is unsupported (I mean, I tried anyway, but of course it didn't work).

I'll open an issue for that. Do you want this fixed before merging?

gregorydlogan replied:
Nope, this is a beta so the occasional issue is to be expected.  This would block promotion of Picard to the default ui though.

gregorydlogan reviewed (changes requested):
Very, very minor changes.  Otherwise looks good to me!

Edit: Also, make a note of this in the release notes as well

Arnei committed:
Update Duplicate Workflow Config Json with missing values

The duplicate workflow was missing some configuration values in its
`configuration_panel_json`. This adds them back. This does not add back
the field for `startDateTime`, as the admin-ui-picard does not support
workflow config fields of type date (or any derivative thereof) at this
point in time.

Arnei committed:
Add release notes for new admin ui

Arnei committed:
Merge remote-tracking branch 'upstream/develop' into add-new-admin-ui-as-beta

# Conflicts:
#	modules/runtime-info-ui-ng/package-lock.json
#	modules/runtime-info-ui-ng/package.json
